cmd/thanos/receive.go

```diff
@@ -821,6 +831,18 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	cmd.Flag("receive.replication-factor", "How many times to replicate incoming write requests.").Default("1").Uint64Var(&rc.replicationFactor)
 
+	cmd.Flag("receive.per-tenant-limit", "The total number of active series that a tenant is allowed to have within a hashring topology.").Uint64Var(&rc.maxPerTenantLimit)
+
+	cmd.Flag("receive.limit-meta-monitoring.url", "Meta-monitoring URL which is compatible with Prometheus Query API for active series limiting.").Default("http://localhost:9090").URLVar(&rc.metaMonitoringUrl)
+
+	cmd.Flag("receive.limit-meta-monitoring.query", "PromQL Query to execute against meta-monitoring, to get the current number of active series for each tenant, across Receive replicas.").Default("sum(prometheus_tsdb_head_series) by (tenant)").StringVar(&rc.metaMonitoringLimitQuery)
+
+	rc.metaMonitoringHttpClient = extflag.RegisterPathOrContent(
+		cmd,
+		"receive.limit-meta-monitoring.http-client",
+		"YAML file or string with http client configs for meta-monitoring.",
+	)
+
```

Comment:
I would suggest grouping all tags under the same namespace. Something like
```
receive.tenant-limits.series-limit
receive.tenant-limits.meta-monitoring-url
receive.tenant-limits.meta-monitoring-query
receive.tenant-limits.meta-monitoring-client
```

Reply:
> receive.tenant-limits.series-limit

I would suggest to name this something like `receive.tenant-limits.max-active-series`. Reading only `series-limit` doesn't tell me they are active series.

Reply:
Agreed, sounds much better.